pkg/utils/jolokia/jolokia.go

```diff
@@ -94,6 +113,16 @@ func (j *Jolokia) getClient() *http.Client {
 			InsecureSkipVerify: true,
 			ServerName:         j.ip,
 		}
+		if j.restricted {
+			httpClientTransport.TLSClientConfig.InsecureSkipVerify = false
+			httpClientTransport.TLSClientConfig.GetClientCertificate =
+				func(cri *tls.CertificateRequestInfo) (*tls.Certificate, error) {
+					return common.GetOperatorClientCertificate(j.client, cri)
+				}
+		}
+		if rootCas, err := common.GetRootCAs(j.client); err == nil {
+			httpClientTransport.TLSClientConfig.RootCAs = rootCas
```

Comment:
Could the jolokia client only trust the operand certificate?

Reply:
I think that depends on the content of the trust bundle, if there is no bundle then I think the OS defaults are in play. I am thinking that the operator has a domain of trust that is gets from the bundle, that domain can be as narrow or as wide as necessary.

Reply:
the implication with self signed certs would be that the bundle needs to be updated for each CR deployed. But using a shared self signed issuer can avoid that.